Lombiq.NodeJs.Extensions/scripts/add-dev-dependencies.js

```diff
@@ -6,22 +6,53 @@ const { EOL } = require('os');
 
 const { handleErrorObject } = require('./handle-error');
 
+const currentDevDependencies = JSON.parse(fs.readFileSync('package.json')).devDependencies;
+function isGreaterThanCurrent([name, version]) {
+    if (!currentDevDependencies) return true;
+
+    const splitVersion = (text) => (typeof text !== 'string' || !text.match(/^[0-9]/)
+        ? null
+        : text.split('.').map((item) => parseInt(item, 10)));
+
+    const newVersion = splitVersion(version);
+    const oldVersion = splitVersion(currentDevDependencies[name]);
+
+    if (newVersion === null) return false;
+    if (oldVersion === null) return true;
+
+    for (let i = 0; i < newVersion.length; i++) {
+        if (newVersion[i] > oldVersion[i] || oldVersion[i] === undefined) return true;
+        if (Number.isNaN(newVersion[i]) || Number.isNaN(oldVersion[i])) return version > currentDevDependencies[name];
+    }
+
+    return false;
+}
+
 const configPath = path.resolve(__dirname, '..', 'package.json');
 const nxDevDependencies = JSON.parse(fs.readFileSync(configPath)).devDependencies;
 const eslintPackages = Object
     .entries(nxDevDependencies)
-    .filter(([name]) => name.startsWith('eslint'))
+    .filter(([name, version]) => name !== 'jest' && version)
```

Comment:
Why is jest excluded?

Reply:
Jest is used [for testing](https://github.com/Lombiq/NodeJs-Extensions/blob/e941e63e0d96a6fb805d9971210ff7695e801275/Lombiq.NodeJs.Extensions/package.json#L83), it's not a dependency we have to inherit. This way we can add other packages too. I used this to [add semver](https://github.com/Lombiq/NodeJs-Extensions/commit/fc934f7f874c49b8d0c878918e656f1cf537b470), but in the end I found that semver [wasn't a right fit for my needs](https://github.com/Lombiq/NodeJs-Extensions/pull/58/files#diff-fc8d3f2f7941a60a5eafd9906e253ec01fb12451fd98caef357a22852b582312R14-R19) so I reverted the package but kept this change because it seems more in spirit for what we want with this script.

Reply:
Ah, clever, thanks.